src/libraries/Microsoft.PowerFx.Core/Public/Types/FormulaType.cs

```diff
@@ -51,6 +52,8 @@ public abstract class FormulaType
 
         public static FormulaType Unknown { get; } = new UnknownType();
 
+        public static FormulaType Deferred { get; } = new DeferredType();
```

Comment:
>Deferred [](http://example.com/codeflow?start=34&length=8)

Can we verify a custom function can't attempt to return one of these? 

More broadly, the challenge  is to ensure that anywhere these deferred types first "Show up" in an expression, we trap it and issue a warning. 

FirstName node is the most obvious place. 
DottedName node is a 2nd spot (since somebody may try to pass it in via a type (can we just fail this at cosntruction?)

CustomFunctions is another possible entry point.

Reply:
Now Aggregate type fails at construction, Custom Function would not be an issue as there is no `DeferredValue` define.

Reply:
That's not entirely true. `ReflectionFunction` has a constructor that can specify return types.

Reply:
Thanks for pointing that out, I have updated it accordingly.